File: 13. Roman to Integer/13. Roman to Integer.java
class Solution {
    public int romanToInt(String s) {
        HashMap<String,Integer> t = new HashMap<String,Integer>();
        HashMap<String,String> m = new HashMap<String,String>();
        t.put("I",1);
        t.put("V",5);
        t.put("X",10);
        t.put("L",50);
        t.put("C",100);
        t.put("D",500);
        t.put("M",1000);
        m.put("IV",";4;");
        m.put("IX",";9;");
        m.put("XC",";90;");
        m.put("XL",";40;");
        m.put("CD",";400;");
        m.put("CM",";900;");
        int res = 0;
        for(Map.Entry<String,String> e: m.entrySet()){
            s = s.replace(e.getKey(),";"+e.getValue()+";");
        }
        String[] sp = s.split(";");
        for(String ms:sp){
            if(ms.length()!=0){
                try{
                    res += Integer.parseInt(ms);
                }catch(NumberFormatException e){
                    for(int i=0;i<ms.length();i++){
                        res += t.get(ms.substring(i,i+1));
                    }
                }
            }
        }
        return res;
    }
}

File: 13. Roman to Integer/13. Roman to Integer.py
class Solution:
    def romanToInt(self, s: str) -> int:
        m = {
            'IV':';4;',
            'IX':';9;',
            'XC':';90;',
            'XL':';40;',
            'CD':';400;',
            'CM':';900;'
        }
        t = {
            'I':1,
            'V':5,
            'X':10,
            'L':50,
            'C':100,
            'D':500,
            'M':1000
            
        }
        for key in m:
            s = s.replace(key,m[key])
        
        s = s.split(';')
        res = 0
        for i in s:
            if i:
                try:
                    res += int(i)
                except:
                    for l in i:
                        res += t[l]
        return res